### DISCUSSION TITLE
Newbie here: is installing kernelSU on Samsung S22 possible?

### DISCUSSION THREAD

**Discussion by iacchi** [tiann/KernelSU#2288] [Q&A]
Hello,
thank you in advance for all your help. I need to install kernelSU on my daily driver, a Samsung Galaxy S22 (Exynos) with original rom, and I want to ask a couple of questions before I proceed.

First of all, I'm currently rooted with Magisk Canary, but a banking app that I really need detects it even if it's hidden and I pass strong integrity. I've read that people got it working with kernelSU instead, hence the switch that I want to do. Here are my questions:

1. Is kernelSU installable on my phone configuration without building a custom kernel? I've installed the app on the phone and it looks like it should be possible, but I would like some reassurance on this regard.

2. All your guides say that I should patch boot.img and then flash it, but Samsung original firmwares work a bit differently afaik. I don't have a boot.img file, but I have an AP tar containing the whole system and the boot.img file in some form instead. I can feed Magisk this AP tar and Magisk Manager is capable of patching it - can the kernelSU app do the same?

3. I'm currently with an about 1 year old firmware installed on the phone (I never had the time to update it), but I was going to update to the latest firmware in all this change process. Can I safely use the latest firmware or is kernelSU going to have issues with it? Which version should I use in case?

4. As I said, I need to install kernelSU and remove Magisk from the system. I have a feeling that installing kernelSU with Magisk installed might be easier as I can get root for it and possibly patch the system in place without needing to patch and flash any files. Is it safe to do it this way, and then remove Magisk via its manager after installing kernelSU? Also re question 2, how would future updates work? Samsung disables OTAs when the bootloader is unlocked, so my only way to update the firmware is to flash it via Odin. Will I be able to patch the files in the future and then flash them with Odin?

Sorry for all the questions, but as I said it's my primary phone and I don't want to mess it up. I also feel this may help others in the future.

**Comment by instawirelessa** [comment]
Installing KernelSU on a Samsung Galaxy S22
Installing KernelSU on a [Samsung Galaxy](https://instawireless.com/collections/samsung-1) S22 is possible but comes with specific challenges and requirements. KernelSU is a tool that enables root access at the kernel level, but achieving this depends on a few key factors related to the device's bootloader and firmware.
Requirements and Considerations
Bootloader Unlocking
•	To install KernelSU, you need an unlocked bootloader.
•	Samsung devices sold in certain regions (e.g., the US) often have bootloaders that cannot be unlocked due to carrier restrictions. International models (e.g., European or Asian variants) typically allow bootloader unlocking.
Custom Kernel Support
•	KernelSU requires a custom kernel that integrates support for KernelSU.
•	You would need to find a compatible custom kernel for the Samsung Galaxy S22 that includes KernelSU support or build one yourself if you have the technical expertise.
Rooting Tools and Methods
•	Rooting the device is a prerequisite for KernelSU installation. Tools like Magisk are commonly used to gain root access.
•	Ensure that the firmware version you're on is compatible with the rooting and custom kernel tools available.
TWRP or Odin
•	For Samsung devices, Odin is often used to flash custom kernels, recovery images (like TWRP), or firmware.
•	Having TWRP installed can make the process of managing and flashing KernelSU-supported kernels easier.
Warranty and Risks
Unlocking the bootloader and installing custom kernels will void your device’s warranty and could lead to issues such as boot loops or a bricked device if not done carefully.
Steps Overview
Unlock the Bootloader
•	Enable Developer Options on your Galaxy S22 and allow OEM unlocking.
•	Use the appropriate method to unlock the bootloader (specific to your model and region).
Root the Device
•	Flash a patched boot image (using Magisk) via Odin or another compatible tool to root the device.
Install a Kernel with KernelSU Support
•	Download or compile a custom kernel for the Galaxy S22 that includes KernelSU support.
•	Flash the kernel using a custom recovery like TWRP or Odin.
Install KernelSU Manager
After the custom kernel is installed, use the KernelSU Manager app to manage root access and permissions.
Important Notes
•	Ensure that the kernel is specifically designed for the Galaxy S22 and matches your firmware version.
•	Back up all important data before proceeding, as these modifications can wipe your device.
•	Check XDA Developers forums or similar communities for device-specific guides and kernel builds for the Samsung Galaxy S22.
If you are unsure or unfamiliar with these processes, proceed with caution or consult with experienced users in your device's developer community.

**Reply by KiwifruitDev** [reply]
This comment is spam, the text was generated with ChatGPT and links to a fishy website.

**Comment by ckgal** [comment]
I am also in the midst of trying to root Samsung S20FE using KernelSU.
Unfortunately, i can't find any updated custom kernel that suits my device, hence i am trying to customize one using KernelSU.
To answer your questions:

_1. Is kernelSU installable on my phone configuration without building a custom kernel? I've installed the app on the phone and it looks like it should be possible, but I would like some reassurance on this regard._

You can try to download the KernelSU Manager APK in https://github.com/tiann/KernelSU/releases to check whether your phone is using GKI and supports KernelSU installation. 
(IMHO, Samsung devices are mostly not GKI, same for my S20FE and custom kernel is required.)

_2. All your guides say that I should patch boot.img and then flash it, but Samsung original firmwares work a bit differently afaik. I don't have a boot.img file, but I have an AP tar containing the whole system and the boot.img file in some form instead. I can feed Magisk this AP tar and Magisk Manager is capable of patching it - can the kernelSU app do the same?_

i supposed KernelSU is root at system level and Magisk is root at boot partition level. (https://kernelsu.org/guide/difference-with-magisk.html) There are similarities and differences but if you do both, i believed the bank app will still be able to detect traces of Magisk which is not what you wanted. 
A more straightforward way instead of using KernelSU, you can try Magisk + HideMyAppList.

_3. I'm currently with an about 1 year old firmware installed on the phone (I never had the time to update it), but I was going to update to the latest firmware in all this change process. Can I safely use the latest firmware or is kernelSU going to have issues with it? Which version should I use in case?_

It is good the check what is your firmware (Settings > About Phone > Software Information > Build) and download either from Samsung Open Source or samfw.com. In case all things failed, you have the original ROM to flash via Odin Tool.

Hope it helps.
//지영

**Comment by mamzolovmv-wq** [comment]
У меня получилось установить на S26 Ultra
